tests/integration/serializers/rest-serializer-test.js

```diff
@@ -143,6 +143,35 @@ test("normalizeResponse with custom modelNameFromPayloadKey", function(assert) {
   });
 });
 
+test("normalizeResponse with type and custom modelNameFromPayloadKey", function(assert) {
+  assert.expect(1);
+
+  env.restSerializer.modelNameFromPayloadKey = function(root) {
+    return Ember.String.singularize(root);
+  };
+
+  var jsonHash = {
+    "home-planets": [{ id: "1", name: "Umber", type: "home-planets" }]
```

Comment:
What do you think of changing `type: "home-planets"` to `type: "my-custom-type"` to make it more clear that `modelNameFromPayloadKey` is used to resolve the model and serializer?